### ISSUE TITLE
Consider different hashing algorithms?

### ISSUE THREAD

**Issue by jimhester** [jeffreyhorner/R-Array-Hash#1]
While mumurhash3 is a very fast hashing algorithm with good collision characteristics it is vulnerable to hash collision attacks ([1](http://www.ocert.org/advisories/ocert-2012-001.html)).  Because of this many languages with built-in hashing are using [SipHash](https://131002.net/siphash/), including Ruby, Perl, Python and Rust.

If the security issue is not a major concern as R is not often exposed to untrusted input [xxHash](https://github.com/Cyan4973/xxHash) is a more recent hashing algorithm than murmurhash3 with better performance characteristics while still passing the SMHasher Test suite.

I believe all three have a similar API, so the change to either of the above should not be terribly time consuming to implement.

**Comment by jeffreyhorner** [comment]
Yes, that is a concern and there may be some speed gains in switching. But the bigger bottlenecks are the garbage collector and the string table.

**Comment by davharris** [comment]
I don't know how safe it is to assume that R users don't work with untrusted inputs. Lots of data sets are downloaded from the web and/or include user-submitted contents from web forms, for instance.
